tests/Microsoft.Identity.Test.Common/Core/Helpers/TokenCacheExtensions.cs

```diff
@@ -14,10 +15,15 @@ internal static class TokenCacheExtensions
     {
         public static void AddAccessTokenCacheItem(this ITokenCacheInternal tokenCache, MsalAccessTokenCacheItem accessTokenItem)
         {
-            lock (tokenCache.LockObject)
+            tokenCache.Semaphore.Wait();
```

Comment:
Wondering if we could do more in terms of unit testing. For example, we could setup some 10 apps do auth (i.e. write to cache), then read the cache, then delete accounts. Then start the apps in parallel and let them work. Or maybe an integration test with an ASP website that fires requests in parallel.

Reply:
Needs to have appropriate testing in place before this goes into the repo

Reply:
Absolutely.  Agree with you and Bogdan on this needing more testing.  This PR was to show the change and get architectural review (thus the "do not merge") and see how existing functionality runs against the central build/test server.  I'll work on adding some tests and a stress threading test on the cache before this is ready for final review/submission.